File: backend/companies-queue/src/main/java/com/datapublica/companies/model/scheduler/TriggerInfo.java
package com.datapublica.companies.model.scheduler;

/**
 *
 */
public class TriggerInfo {
    public TriggerType type = TriggerType.CUSTOM;
    public String expression;
    /**
     * Periods are in seconds
     */
    public long period;
    public long initialDelay;

    public TriggerInfo(TriggerType type) {
        this.type = type;
    }

    public TriggerInfo(String cronExpression) {
        type = TriggerType.CRON;
        expression = cronExpression;
    }

    public TriggerInfo(long period, long initialDelay) {
        this.period = period;
        this.initialDelay = initialDelay;
        this.type = TriggerType.RATE;
    }

    public TriggerInfo(long period, long initialDelay, boolean fixed) {
        this(period, initialDelay);
        if(fixed)
            this.type = TriggerType.FIXED_RATE;
    }

    public TriggerInfo() {
    }
}

File: front/frontend-admin/app/api/TriggerInfo.ts
/// <reference path="api.d.ts" />

namespace api {
    'use strict';

    export interface TriggerInfo {

        "expression"?: string;

        "initialDelay"?: number;

        "period"?: number;

        "type"?: TriggerInfo.TypeEnum;
    }

    export namespace TriggerInfo {

        export enum TypeEnum { 
            CRON = <any> 'CRON',
            RATE = <any> 'RATE',
            FIXED_RATE = <any> 'FIXED_RATE',
            CUSTOM = <any> 'CUSTOM'
        }
    }
}
